dali/pipeline/executor/executor.h

```diff
@@ -478,15 +480,19 @@ void Executor<WorkspacePolicy, QueuePolicy>::ShareOutputs(DeviceWorkspace *ws) {
   if (exec_error_ || QueuePolicy::IsStopSignaled()) {
     // TODO(klecki) collect all errors
     std::lock_guard<std::mutex> errors_lock(errors_mutex_);
-    std::string error = errors_.empty() ? "Unknown error" : errors_.front();
+    std::string error = errors_.empty() ? QueuePolicy::IsStopSignaled() && !exec_error_
+                                            ? "Stop signaled" : "Unknown error"
+                                        : errors_.front();
```

Comment:
I was never a fan of nested ternary operators, I would pack this up and use twice:

```
std::string GetErrorMessageHelper() {
  if (!errors_.empty()) { 
    return errors_.front();
  }
  if (QueuePolicy::IsStopSignaled() && !exec_error_) {
    return "Stop signaled";
  }
  return "Unknown error";
}
```

Reply:
I'm all for extracting it to a function, but ternary operators vs if/else is a matter of personal taste.